### DISCUSSION TITLE
Support CDC-ACM for ESP32C3

### DISCUSSION THREAD

**Discussion by matsujirushi** [nanoframework/Home#1215] [Q&A] [answered]
Hi,

I'm using nanoFramework on Seeed Studio XIAO ESP32C3.
This requires an additional USB-UART converter.

Are there any plans to use the ESP32C3's built-in USB serial/JTAG controller to communicate with Visual Studio?

![image](https://user-images.githubusercontent.com/8079404/213916086-1779c27e-3469-4769-b5df-e15064d5c3ac.png)

https://github.com/nanoframework/nf-interpreter/blob/main/targets/ESP32/_common/WireProtocol_HAL_Interface.c

**Comment by josesimoes** [comment, marked as answer]
This is kind of the opposite (and the same) as it's happening with S2 right now. The build for that one is configured to use the hardware CDC and there are targets using an external serial to USB chip. There's already an issue raised for this one #890 better open a new one specific for the C3. Care to do that please?

**Reply by matsujirushi** [reply]
Thank you for your reply.
The same thing is happening with the ESP32S2. I did not know.

I think there are the following issues to make it work with the ESP32C3 built-in USB serial/JTAG controller.

1. How to switch between UART and built-in USB serial/JTAG controller firmware?
2. We have to create a driver for the built-in USB serial/JTAG controller. (I don't think it can be diverted because it is different from ESP32S2.)

I think that No.1 can be switched by specifying target.
**Do you have any plans to make No.2?**

**Reply by josesimoes** [reply]
Yes, there are plans to provide different firmware images with and without use of internal USB CDC. 
For S2 and C3.
